spring-integration-mail/src/main/java/org/springframework/integration/mail/AbstractMailReceiver.java

```diff
@@ -377,7 +443,7 @@ private void setMessageFlags(Message[] filteredMessages) throws MessagingExcepti
 				filteredMessages.add(message);
 			}
 		}
-		return filteredMessages.toArray(new Message[filteredMessages.size()]);
+		return filteredMessages.toArray(new MimeMessage[filteredMessages.size()]);
```

Comment:
My IDEA would like to see it with the previous state.
Since we do the explicit message cast anyway, I'd prefer to see the `filteredMessages` as a `List<MimeMessage>`, to avoid that noise from my IDE.
We may even have some `warn` from the compiler on the matter...